lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -895,7 +895,7 @@ def add_auth_wallet(xml, payment_method, options)
         brand = card_brand(payment_method).to_sym
 
         case brand
-        when :visa
+        when :visa, :discover
           xml.tag! 'ccAuthService', { 'run' => 'true' } do
             xml.tag!('cavv', payment_method.payment_cryptogram) unless commerce_indicator
             xml.commerceIndicator commerce_indicator.nil? ? ECI_BRAND_MAPPING[brand] : commerce_indicator
```

Comment:
the XID thing, which im not sure happens in our circumstance here might be an only VISA thing, so maybe the correct discover code might be the same code, but without the XID line